src/models/activity.rs

```diff
@@ -284,9 +286,18 @@ impl Activity {
     ) -> Vec<Activity> {
         let mut activities: Vec<Activity> = vec![];
 
-        let compatible_hours_overlay =
-            Self::get_compatible_overlay_hours(goal_to_use, calendar, time_budget);
-
+        let compatible_hours_overlay = Activity::get_compatible_hours_overlay(
```

Comment:
Why is this code refactored to code duplication?

Reply:
See https://github.com/tijlleenders/ZinZen-scheduler/pull/446